### PR TITLE
feat(cat-voices): recover flow scaffloding

### LINKED ISSUE #950: 🛠️ [TASK] : Restore Flow Stages Placeholders
Part of #821

### LINKED ISSUE #951: 🛠️ [TASK] : Recover Methods
Part of #821

### LINKED ISSUE #952: 🛠️ [TASK] : SeedPhrase instructions
Part of #821

### PR DESCRIPTION
**Pull request by damian-molinski**
# Description

- Recover methods and panels scaffoling 
- Extracts `VoicesIndicator` out of `VoicesErrorIndicator`
- Extracts `RegistrationTile`
- First two panels `RecoverMethodPanel` and `SeedPhraseInstructionsPanel`.

## Related Issue(s)

- Closes #950 
- Closes #951 
- Closes #952 

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

File: catalyst_voices/packages/catalyst_voices_blocs/lib/src/registration/state_data/recover_state_data.dart [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import 'package:equatable/equatable.dart';
+
+final class RecoverStateData extends Equatable {
+  const RecoverStateData({
+    this.foundKeychain = false,
+  });
+
+  final bool foundKeychain;
```
Comment (dtscalac) [inline catalyst_voices/packages/catalyst_voices_blocs/lib/src/registration/state_data/recover_state_data.dart:8]: fields above constructors please

File: catalyst_voices/lib/widgets/indicators/voices_success_indicator.dart [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import 'package:catalyst_voices/widgets/widgets.dart';
+import 'package:catalyst_voices_assets/catalyst_voices_assets.dart';
+import 'package:flutter/material.dart';
+
+/// A generic error state with optional retry button.
+class VoicesSuccessIndicator extends StatelessWidget {
+  /// The description of the error.
+  final String message;
+
+  /// Usually [VoicesTextButton]. Will be appended at the end of widget
```
Comment (dtscalac) [inline catalyst_voices/lib/widgets/indicators/voices_success_indicator.dart:10]:
```suggestion
  /// Usually [VoicesTextButton]. Will be appended at the end of widget.
```

File: catalyst_voices/lib/widgets/indicators/voices_error_indicator.dart [hunk 3]
```diff
@@ -22,37 +21,17 @@ class VoicesErrorIndicator extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    return Container(
-      padding: const EdgeInsets.symmetric(vertical: 6, horizontal: 16),
-      decoration: BoxDecoration(
-        border: Border.all(
-          color: Theme.of(context).colors.outlineBorderVariant!,
-          width: 1,
+    return VoicesIndicator(
```
Comment (dtscalac) [inline catalyst_voices/lib/widgets/indicators/voices_error_indicator.dart:24]: You're sure we want to use this in all the places? This design was supposed to be implemented by VoicesErrorIndicator: https://www.figma.com/design/80aDAQgoD0v06DO2Sb76ok/Catalyst--Minimal-Valuable-Experiments-(MVE's)?node-id=6642-93604&t=AdFNu1p5NB0czRyL-4

**Review by dtscalac — CHANGES_REQUESTED** [review]

File: catalyst_voices/lib/widgets/indicators/voices_error_indicator.dart (hunk 3 above)
Comment (damian-molinski) [reply catalyst_voices/lib/widgets/indicators/voices_error_indicator.dart:24]: Yes. Niels just re-used it [here](https://www.figma.com/design/80aDAQgoD0v06DO2Sb76ok/Catalyst--Minimal-Valuable-Experiments-(MVE's)?node-id=7058-45265&m=dev)

Comment (dtscalac) [reply catalyst_voices/lib/widgets/indicators/voices_error_indicator.dart:24]: Okay, does it still look the same after replacing the previous implementation by VoicesIndicator? The last time i checked voices indicator looked much differently. Can you add a screenshot with error state to the issue description please?

Comment (dtscalac) [reply catalyst_voices/lib/widgets/indicators/voices_error_indicator.dart:24]: Alright, I got it wrong, VoicesIndicator is a new widget and I thought it looks like VoicesStatusIndicator which is very different. No issue then, thanks 👍

**Review by dtscalac — APPROVED** [review]
LGTM

**Pull request merged**